# Lab 8 Exercise 7

## Method with parameter and return value

1. สร้าง console application project

```
    dotnet new console --name Lab08_Ex07
```

2. เปลี่ยน code ให้เป็นดังต่อไปนี้

> ```csharp
>  1. var adder = new MyProgram();
>  2. int x = 2, y = 3;
>  3. int add_result = adder.add(x, y);
>  4. System.Console.WriteLine($"{x} + {y} = {add_result}");
>  5. 
>  6. int i = 2, j = 3;
>  7. int multiply_result = adder.multiply(i, j);
>  8. System.Console.WriteLine($"{i} x {j} = {multiply_result}");
>  9. 
> 10. class MyProgram
> 11. {
> 12.     public int add(int a, int b)
> 13.     {
> 14.         return a + b;
> 15.     }
> 16.     public int multiply(int a, int b)
> 17.     {
> 18.         return a * b;
> 19.     }
> 20. }
> ```



3. Build project โดยการใช้คำสั่ง

```
    dotnet build  Lab08_Ex07
```

4. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 3



*คำตอบ*

Build succeeded in 1.2s

5. Run project โดยการใช้คำสั่ง

```
    dotnet run --project Lab08_Ex07
```

6. บันทึกผลที่ได้จากการรันคำสั่งในข้อ 5





*คำตอบ*

2 + 3 = 5

2 x 3 = 6

7. อธิบายสิ่งที่พบในการทดลอง



*คำตอบ*

โค้ด error error CS1002: ; expected
error CS1519: Invalid token '.' in class, record, struct, or interface member declaration
error CS1022: Type or namespace definition, or end-of-file expected
และแก้ไขให้เรียบร้อย


8. นำ code ในข้อ 2 มารันบน polyglot notebook  โดยใส่ใน  code cell ด้านล่างนี้


In [ ]:
// พื้นที่เขียน code ตามข้อ 2
var adder = new MyProgram();
 int x = 2, y = 3;
 int add_result = adder.add(x, y);
 System.Console.WriteLine($"{x} + {y} = {add_result}");
 
 int i = 2, j = 3;
 int multiply_result = adder.multiply(i, j);
 System.Console.WriteLine($"{i} x {j} = {multiply_result}");
 
 class MyProgram
 {
     public int add(int a, int b)
     {
        return a + b;
    }
    public int multiply(int a, int b)
     {
         return a * b;
     }
 }

2 + 3 = 5
2 x 3 = 6


8.1 ผลการรันเป็นอย่างไร



*คำตอบ*

2 + 3 = 5
2 x 3 = 6

8.2 เหมือนหรือต่างจาก console application อย่างไร

*คำตอบ*

เหมือนกันทั้งคู่

## คำถาม

1. ให้เขียน method ที่รับ parameter เป็น string และส่งกลับค่าความยาวของ string นั้น

In [ ]:
// พื้นที่เขียน code และทดสอบ
public static int GetStringLength(string inputString)
{
  if (inputString == null)
  {
    return 0; 
  }
  return inputString.Length;
}
string myString = "Hello, world!";
int length = GetStringLength(myString);
Console.WriteLine("ความยาวของสตริงคือ: " + length); 

string nullString = null;
int nullLength = GetStringLength(nullString);
Console.WriteLine("ความยาวของสตริง null คือ: " + nullLength); 

ความยาวของสตริงคือ: 13
ความยาวของสตริง null คือ: 0


*คำตอบ*

ความยาวของสตริงคือ: 13
ความยาวของสตริง null คือ: 0

2. ให้สร้างโปรแกรมสูตรคูณ โดยประยุกต์ใช้เมธอด  `public int multiply(int a, int b)`


In [ ]:
// พื้นที่เขียน code และทดสอบ
class MyProgram
{
    
    public int Multiply(int a, int b)
    {
        return a * b;  
    }
}

   var calculator = new MyProgram();

        int number = 9;

        for (int i = 1; i <= 12; i++)
        {
            int result = calculator.Multiply(number, i); 
            Console.WriteLine($"{number} x {i} = {result}");
        }

9 x 1 = 9
9 x 2 = 18
9 x 3 = 27
9 x 4 = 36
9 x 5 = 45
9 x 6 = 54
9 x 7 = 63
9 x 8 = 72
9 x 9 = 81
9 x 10 = 90
9 x 11 = 99
9 x 12 = 108


*คำตอบ*

9 x 1 = 9
9 x 2 = 18
9 x 3 = 27
9 x 4 = 36
9 x 5 = 45
9 x 6 = 54
9 x 7 = 63
9 x 8 = 72
9 x 9 = 81
9 x 10 = 90
9 x 11 = 99
9 x 12 = 108

3. จากโปรแกรมในคำถามข้อที่ 2 ให้สร้างเป็นสูตรคูณ แม่ 2 - 12

In [ ]:
// พื้นที่เขียน code และทดสอบ
class MyProgram
{
    
    public int Multiply(int a, int b)
    {
        return a * b;  
    }
}
    var calculator = new MyProgram();  

        
        for (int number = 2; number <= 12; number++)
        {
            Console.WriteLine($"Multiplication table of {number}:");
            for (int i = 1; i <= 12; i++)
            {
                int result = calculator.Multiply(number, i); 
                Console.WriteLine($"{number} x {i} = {result}");
            }
            Console.WriteLine();  
        }

Multiplication table of 2:
2 x 1 = 2
2 x 2 = 4
2 x 3 = 6
2 x 4 = 8
2 x 5 = 10
2 x 6 = 12
2 x 7 = 14
2 x 8 = 16
2 x 9 = 18
2 x 10 = 20
2 x 11 = 22
2 x 12 = 24

Multiplication table of 3:
3 x 1 = 3
3 x 2 = 6
3 x 3 = 9
3 x 4 = 12
3 x 5 = 15
3 x 6 = 18
3 x 7 = 21
3 x 8 = 24
3 x 9 = 27
3 x 10 = 30
3 x 11 = 33
3 x 12 = 36

Multiplication table of 4:
4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36
4 x 10 = 40
4 x 11 = 44
4 x 12 = 48

Multiplication table of 5:
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50
5 x 11 = 55
5 x 12 = 60

Multiplication table of 6:
6 x 1 = 6
6 x 2 = 12
6 x 3 = 18
6 x 4 = 24
6 x 5 = 30
6 x 6 = 36
6 x 7 = 42
6 x 8 = 48
6 x 9 = 54
6 x 10 = 60
6 x 11 = 66
6 x 12 = 72

Multiplication table of 7:
7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63
7 x 10 = 70
7 x 11 = 77
7 x 12 = 84

Multiplication table of 8:
8 x 1 =

*คำตอบ*

Multiplication table of 2:
2 x 1 = 2
2 x 2 = 4
2 x 3 = 6
2 x 4 = 8
2 x 5 = 10
2 x 6 = 12
2 x 7 = 14
2 x 8 = 16
2 x 9 = 18
2 x 10 = 20
2 x 11 = 22
2 x 12 = 24

Multiplication table of 3:
3 x 1 = 3
3 x 2 = 6
3 x 3 = 9
3 x 4 = 12
3 x 5 = 15
3 x 6 = 18
3 x 7 = 21
3 x 8 = 24
3 x 9 = 27
3 x 10 = 30
...
12 x 10 = 120
12 x 11 = 132
12 x 12 = 144

Output is truncated. View as a scrollable element or open in a text editor. Adjust cell output settings...